javalib/src/main/scala/java/util/package.scala

```diff
@@ -29,7 +31,180 @@ package object util {
   private[util] def defaultOrdering[E]: Ordering[E] = {
     new Ordering[E] {
       def compare(a: E, b: E): Int =
-        a.asInstanceOf[Comparable[E]].compareTo(b)
+        if (a === null) {
+          if (b === null) 0
+          else -1
+        } else {
+          a.asInstanceOf[Comparable[E]].compareTo(b)
+        }
+    }
+  }
+
+  private[util] class NavigableView[E](original: NavigableSet[E], inner: () => mutable.SortedSet[Box[E]])
+      extends AbstractCollection[E] with NavigableSet[E] with SortedSet[E] {
+
+    def size(): Int =
+      iterator.size
+
+    override def contains(o: Any): Boolean =
+      inner().contains(Box(o.asInstanceOf[E]))
+
+    override def add(e: E): Boolean =
+      original.add(e)
+
+    override def remove(o: Any): Boolean = {
+      if (contains(o))
+        original.remove(o)
+      else false
+    }
+
+    def iterator(): Iterator[E] =
+      inner().iterator.map(_.inner)
```

Comment:
You need to propagate the calls from remove to `original` just like you did with the `descendingIterator`.